windows_service/tests/test_windows_service.py

```diff
@@ -5,32 +5,52 @@
 
 
 def test_bad_config(check, instance_bad_config):
+    c = check(instance_bad_config)
     with pytest.raises(ValueError):
-        check.check(instance_bad_config)
+        c.check(instance_bad_config)
 
 
 def test_basic(aggregator, check, instance_basic):
-    check.check(instance_basic)
+    c = check(instance_basic)
+    c.check(instance_basic)
     aggregator.assert_service_check(
-        check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:EventLog', 'optional:tag1'], count=1
+        c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:EventLog', 'optional:tag1'], count=1
     )
     aggregator.assert_service_check(
-        check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:Dnscache', 'optional:tag1'], count=1
+        c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:Dnscache', 'optional:tag1'], count=1
     )
     aggregator.assert_service_check(
-        check.SERVICE_CHECK_NAME, status=check.CRITICAL, tags=['service:NonExistentService', 'optional:tag1'], count=1
+        c.SERVICE_CHECK_NAME, status=c.CRITICAL, tags=['service:NonExistentService', 'optional:tag1'], count=1
     )
 
 
 def test_wildcard(aggregator, check, instance_wildcard):
-    check.check(instance_wildcard)
-    aggregator.assert_service_check(check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:EventLog'], count=1)
-    aggregator.assert_service_check(check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:EventSystem'], count=1)
-    aggregator.assert_service_check(check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:Dnscache'], count=1)
+    c = check(instance_wildcard)
+    c.check(instance_wildcard)
+    aggregator.assert_service_check(c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:EventLog'], count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:EventSystem'], count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:Dnscache'], count=1)
 
 
 def test_all(aggregator, check, instance_all):
-    check.check(instance_all)
-    aggregator.assert_service_check(check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:EventLog'], count=1)
-    aggregator.assert_service_check(check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:Dnscache'], count=1)
-    aggregator.assert_service_check(check.SERVICE_CHECK_NAME, status=check.OK, tags=['service:EventSystem'], count=1)
+    c = check(instance_all)
+    c.check(instance_all)
+    aggregator.assert_service_check(c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:EventLog'], count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:Dnscache'], count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_NAME, status=c.OK, tags=['service:EventSystem'], count=1)
+
+
+@pytest.mark.e2e
+def test_basic_e2e(dd_agent_check, check, instance_basic):
+    aggregator = dd_agent_check(instance_basic)
+
+    c = check(instance_basic)
```

Comment:
do you need an instance of the check just to access those constants?

Reply:
👍 , could probably be `datadog_checks.base.AgentCheck.OK`.

Reply:
Yes, could probably be `datadog_checks.base.AgentCheck.OK`.